warp/_src/builtins.py

```
@@ -9996,22 +10032,22 @@ def fn(arg_types, arg_values):
 )
 add_builtin(
     "mul",
-    input_types={"a": quaternion(dtype=Scalar), "b": Scalar},
-    value_func=scalar_mul_create_value_func(quaternion(dtype=Scalar)),
+    input_types={"a": quaternion(dtype=Float), "b": Scalar},
```

Ruff (0.14.10)

`quaternion` may be undefined, or defined from star imports

(F405)